In [1]:
%matplotlib inline

In [2]:
import numpy
from sourmash_lib import fig
import pylab

In [9]:
!wc -l microbes.2.*txt

     175 microbes.2.k15.txt
      60 microbes.2.k31.txt
      60 microbes.2.k51.sort.txt
      60 microbes.2.k51.txt
     355 total


In [10]:
!sort -k 1 microbes.2.k31.txt | cat -

0.02 NZ_LGKP01000025.1 Herpetosiphon geysericola strain DSM 7119 contig_1, whole genome shotgun sequence
0.06 NC_013887.1 Methanocaldococcus sp. FS406-22, complete genome
0.37 NC_010483.1 Thermotoga sp. RQ2, complete genome
0.45 NZ_CH959311.1 Sulfitobacter sp. EE-36 scf_1099451318005 genomic scaffold, whole genome shotgun sequence
0.47 NC_009665.1 Shewanella baltica OS185, complete genome
0.74 NC_013968.1 Haloferax volcanii DS2 plasmid pHV1, complete sequence
0.75 NZ_ABZS01000228.1 Sulfurihydrogenibium yellowstonense SS-5 gcontig_18, whole genome shotgun sequence
0.87 NC_003911.12 Ruegeria pomeroyi DSS-3, complete genome
0.88 NC_003272.1 Nostoc sp. PCC 7120 DNA, complete genome
0.88 NC_005835.1 Thermus thermophilus HB27, complete genome
0.88 NZ_CH959317.1 Sulfitobacter sp. NAS-14.1 scf_1099451320472 genomic scaffold, whole genome shotgun sequence
0.94 NC_001263.1 Deinococcus radiodurans R1 chromosome 1, complete sequence
0.94 NC_008751.1 Desulfovibrio vulgaris DP4, complete genome
0.94

### Organisms present in metagenome but not found with SBT search
- Herpetosiphon aurantiacus 
- Methanocaldococcus jannaschii
- Methanopyrus kandleri 
- Nanoarchaeum equitans 
- Pyrobaculum aerophilum IM2
- Pyrobaculum arsenaticum 
- Pyrobaculum calidifontis 
- Thermus thermophilus HB8

### Organisms found with sbt search but not present in metagenome (Fraction of genome, Refseq ID, organism)
- 0.02 NZ_LGKP01000025.1 Herpetosiphon geysericola strain DSM 7119 contig_1, whole genome shotgun sequence
- 0.06 NC_013887.1 Methanocaldococcus sp. FS406-22, complete genome
- 0.88 NC_005835.1 Thermus thermophilus HB27, complete genome




In [12]:
!sourmash compute -k 31,51 --dna --scaled 10000 -n 0 /Users/philliptbrooks/Desktop/sourmash/mircea_v_refseq/*fna.gz

# running sourmash subcommand: compute
computing signatures for files: /Users/philliptbrooks/Desktop/sourmash/mircea_v_refseq/GCF_000007185.1_ASM718v1_genomic.fna.gz, /Users/philliptbrooks/Desktop/sourmash/mircea_v_refseq/GCF_000007225.1_ASM722v1_genomic.fna.gz, /Users/philliptbrooks/Desktop/sourmash/mircea_v_refseq/GCF_000008085.1_ASM808v1_genomic.fna.gz, /Users/philliptbrooks/Desktop/sourmash/mircea_v_refseq/GCF_000015805.1_ASM1580v1_genomic.fna.gz, /Users/philliptbrooks/Desktop/sourmash/mircea_v_refseq/GCF_000016385.1_ASM1638v1_genomic.fna.gz, /Users/philliptbrooks/Desktop/sourmash/mircea_v_refseq/GCF_000018565.1_ASM1856v1_genomic.fna.gz, /Users/philliptbrooks/Desktop/sourmash/mircea_v_refseq/GCF_000091545.1_ASM9154v1_genomic.fna.gz, /Users/philliptbrooks/Desktop/sourmash/mircea_v_refseq/GCF_000091665.1_ASM9166v1_cds_from_genomic.fna.gz
Computing signature for ksizes: [31, 51]
Computing only DNA (and not protein) signatures.
Computing a total of 2 signatures.
Computing signature for

In [19]:
!sourmash sbt_index -k 31 --dna mircea_31 ../signatures/mircea.2.fa.sig

# running sourmash subcommand: sbt_index
loading 1 files into SBT
...sig loading 63
loaded 64 sigs; saving SBT under "mircea_31"


In [24]:
ls -lh ../signatures/mircea.2.fa.sig

-rw-r--r--  1 philliptbrooks  staff   2.3M Feb  8 10:58 ../signatures/mircea.2.fa.sig


In [33]:
!for i in 
do 
sourmash sbt_search -k 31 --dna mircea_31 ../signatures/${i} --threshold=0.001
done

# running sourmash subcommand: sbt_search
usage: sourmash [-h] [-k KSIZE] [--threshold THRESHOLD]
                [--save-matches SAVE_MATCHES] [--best-only] [--protein]
                [--no-protein] [--dna] [--no-dna]
                sbt_name query
sourmash: error: unrecognized arguments: ../signatures/GCF_000007225.1_ASM722v1_genomic.fna.gz.sig ../signatures/GCF_000008085.1_ASM808v1_genomic.fna.gz.sig ../signatures/GCF_000015805.1_ASM1580v1_genomic.fna.gz.sig ../signatures/GCF_000016385.1_ASM1638v1_genomic.fna.gz.sig ../signatures/GCF_000016785.1_ASM1678v1_genomic.fna.gz.sig ../signatures/GCF_000018565.1_ASM1856v1_genomic.fna.gz.sig ../signatures/GCF_000091545.1_ASM9154v1_genomic.fna.gz.sig ../signatures/GCF_000091665.1_ASM9166v1_cds_from_genomic.fna.gz.sig


In [6]:
!sort -k 3 microbes.2.k51.txt | cat -

Composition:
1.00 NC_012483.1 Acidobacterium capsulatum ATCC 51196, complete genome
1.00 NC_013926.1 Aciduliprofundum boonei T469, complete genome
1.00 NC_010655.1 Akkermansia muciniphila ATCC BAA-835, complete genome
1.00 NC_000917.1 Archaeoglobus fulgidus DSM 4304, complete genome
0.99 NC_004663.1 Bacteroides thetaiotaomicron VPI-5482 chromosome, complete genome
0.99 NC_009614.1 Bacteroides vulgatus ATCC 8482, complete genome
1.00 NC_002927.3 Bordetella bronchiseptica strain RB50, complete genome
1.00 NC_007951.1 Burkholderia xenovorans LB400 chromosome 1, complete sequence
0.97 NC_012034.1 Caldicellulosiruptor bescii DSM 6725, complete genome
1.00 NC_009437.1 Caldicellulosiruptor saccharolyticus DSM 8903, complete genome
1.00 NC_010803.1 Chlorobium limicola DSM 245, complete genome
1.00 NC_008639.1 Chlorobium phaeobacteroides DSM 266, complete genome
1.00 NC_009337.1 Chlorobium phaeovibrioides DSM 265, complete genome
1.00 NC_002932.3 Chlorobium tepidum TLS chromosome, complete geno

### Organisms present in metagenome but not found with SBT search
- Herpetosiphon aurantiacus 
- Methanocaldococcus jannaschii
- Methanopyrus kandleri 
- Nanoarchaeum equitans 
- Pyrobaculum aerophilum IM2
- Pyrobaculum arsenaticum 
- Pyrobaculum calidifontis 
- Thermoanaerobacter pseudethanolicus 
- Thermus thermophilus HB8 * 

### Organisms found with sbt search but not present in metagenome (Fraction of genome, Refseq ID, organism)
- 0.02 NZ_LGKP01000025.1 Herpetosiphon geysericola strain DSM 7119 contig_1, whole genome shotgun sequence
- 0.03 NZ_CP009149.1 Methanocaldococcus sp. JH146, complete genome
- 0.91 NC_005835.1 Thermus thermophilus HB27, complete genome *
